### ISSUE TITLE
Azure Storage Explorer Stuck at loading

### ISSUE THREAD

**Issue by vikramhegde123** [microsoft/AzureStorageExplorer#578]
**Storage Explorer Version**: 1.4.0
**Platform/OS Version**: Windows 7
**Architecture**: 
**Regression From**: Not working on any versions

#### Steps to Reproduce: ####
1. https://azure.microsoft.com/en-in/features/storage-explorer/
2. Install Azure Storage Explorer
3. Open the application after installing. 

#### Expected Experience: ####
It should prompt to login in to azure with credentials and should load azure storage window.

#### Actual Experience: ####
Microsoft Azure Storage Explorer is stuck at loading screen.
![azure-storage-explorer-loading-issue](https://user-images.githubusercontent.com/20538645/44074747-88cc6446-9fb7-11e8-860a-25b7679281b9.PNG)

**Comment by MRayermannMSFT** [comment]
@vikramhegde123 , can you take a look in task manager and let me know how many storage explorer processes there are?

**Comment by hawkins-dale** [comment]
I have the same issue!  
In task manager, I see six processes names StorageExplorer.exe.
I am attempting to start Storage Explorer 1.7.0 on Windows 7.

**Comment by sscheider** [comment]
No one has posted a possible solution, so I will post what I did:
1. Open Task Manager
2. Go to the Processes Tab
3. ensure that the Names are being ordered and find Microsoft Azure Storage Explorer (32 bit) ...
4. Right click on the process and select End Task from the context menu
5. Wait for the process to fully shut down (I had a Shutdown/Restart window show up for a while)
6. Restart Storage Explorer.

Mine came up fine at that point. I'd like to know of a root cause and if I can do something to prevent this in the future.
Thanks!

**Comment by mhammons** [comment]
Hi, I too am seeing this problem. 
MASE v1.12.0
Windows 10 Pro 1903

Just downloaded the latest version and it hangs on the startup screen. Previous version was doing the same thing.

**Comment by jodybrewster** [comment]
Latest version is hanging on me as well

MacOS Catalina 10.15.2

Is this an electron app?  Maybe use unhandled for startup issues?
https://github.com/sindresorhus/electron-unhandled

Just a suggestion, I'm a JS dev with some degree of native development if you want me to help debug?

**Comment by darrenstilesag** [comment]
Hi, We are seeing the same issue on multiple machines
MASE v1.12.0
Windows 7

Takes a long time to open, then will eventually crash whilst trying to authenticate.  Eventually reports a .Net Framework Error saying install 4.6.2 or newer although running the latest version 4.8. ran a repair on .Net but no luck.

**Comment by MRayermannMSFT** [comment]
Can y'all please do the following?

# Collect Service Hub Logs

## Clean the ServiceHub logs
1. Open File Explorer and go to C:\Users\your username\AppData\Local\Temp\servicehub and delete the logs folder

## Set Trace Level
1. In the start menu search for path
2. Choose "Edit the system environment variables" image
3. In the dialog that pops up, click "Environment Variables...", and then in the next dialog click the first "New...", and finally name the variable SERVICEHUBTRACELEVEL and set the value to be ALL.
4. Hit Ok's until all the dialogs are closed.

## Repro and get us logs
1. Open Storage Explorer
2. Wait for the dialog to pop up
3. Go back to C:\Users\your username\AppData\Local\Temp\servicehub, zip up the logs folder, and send us the Zip

**Comment by mhammons** [comment]
Hi Matthew,

I don’t have a servicehub folder under temp path:
[cid:image001.jpg@01D5FD36.F1BDD970]

Logs attached.

**Comment by prashanthkul** [comment]
Happening in mac (10.15.3) too. The application does not load. Any recommendation?

**Comment by darrenstilesag** [comment]
Hi Matthew,

Just followed your instructions but nothing has happened.
Storage Explorer tries to load but nothing happens.
The only way I get Storage Explorer to load is to delete .IdentityService folder in AppData/Local

Thanks

Darren

**Comment by MRayermannMSFT** [comment]
@prashanthkul here are similar instructions for you

Make sure MSAL is enabled
1.	Open Storage Explorer
2.	Make sure MSAL is enabled in the settings, if it isn’t, then enable and restart Storage Explorer
3.	Remove all accounts
4.	Close Storage Explorer
 
Clean the ServiceHub logs
1.	Open terminal
2.	`cd /Users/your username/.ServiceHub/logs`
3.	`open .`
4.	In the Finder window that appears, delete everything, leave this finder window Finder open
 
Set Trace Level
1.	Go back to terminal
2.	`export SERVICEHUBTRACELEVEL=ALL` 
3. do not close this terminal
 
Repro and get us logs
Use the same terminal window from the previous section, you may have to modify the `cd` path depending on where you put our .app file.
1.	`cd /Applications/Microsoft\ Azure\ Storage\ Explorer.app/Contents/MacOS`
2. `./Microsoft\ Azure\ Storage\ Explorer.`
3.	Repro the issue
4.	Go back to the Finder window from here, zip up everything in there, and send us the Zip




@darrenstilesag are you saying that deleting that folder unblocked you?
@mhammons your logs didn't make it. You have to come to github.com and attach the files here.

**Comment by darrenstilesag** [comment]
Hi,

Originally, deleting the .identityservice folder worked each time I needed to log in to storage explorer, but more recently that stopped working.  I have since upgraded to v1.11 and now storage explorer is working fine.  I am running windows 7 (I know!) so not sure if this causes some issues not usually seen?

Anyway for me it’s all working and as soon as I get back to the office I’m upgrading to windows 10 so will try and upgrade storage explorer again.

Thanks

Darren

**Comment by MRayermannMSFT** [comment]
@darrenstilesag I'm to hear you're unstuck. 

We don't test against Windows 7, though I can't think of any immediate reasons why we wouldn't work on it. But staying up to date on both our version and Windows (though I can sympathize with many of the various reasons that may not be possible for you) is the best way to avoid unexpected issues..

Anyway, if you have any future problems, feel free to open a new issue. Until then, stay safe!

**Comment by mhammons** [comment]
Hi, just installed MASE 1.13 and it still takes 30 minutes or longer to load my storage accounts. The only change is it no longer hangs on the logo screen, it now opens MASE but now see the spinning icon while it tries to load accounts/subscriptions.

**Comment by mhammons** [comment]
@MRayermannMSFT logs are attached in this message. Still have the problems above.
Do not have a servicehub folder under temp path. See screenshot.
![missing-servicehub-folder](https://user-images.githubusercontent.com/36681041/81950803-6944aa80-95d2-11ea-924c-c1d8ffcb9031.jpg)

[logs.zip](https://github.com/microsoft/AzureStorageExplorer/files/4629213/logs.zip)

**Comment by MRayermannMSFT** [comment]
@mhammons  if you don't have a servicehub folder, then where are these logs from?

**Comment by reddevil7nine** [comment]
This may not apply but it is a change to how the app used to work; maybe it will be helpful.

I've currently got 1.14.2 installed via [scoop](https://scoop.sh/). 

For those that are unfamiliar, scoop extracts the app to a version-named folder and then creates a symbolic link named "current" to that same folder. Running Storage Explorer through the symbolic link path doesn't work (it used to). Running it directly from the version named folder does.

**Comment by MRayermannMSFT** [comment]
@reddevil7nine can you open a specific issue for that? :)

**Comment by MRayermannMSFT** [comment]
We're fixing the scoop issue, but since @mhammons  hasn't replied to my question, I'm going to go ahead and close this. If anyone is still having startup problems, please open a new issue. Thanks.

**Comment by mhammons** [comment]
Sorry...I don't recall where the logs came from...might have been from a previous install. I had deleted and re-installed. 

I now have 1.13 and it now loads a bit slowly, but no where near the issues reported earlier. So as of now, it's working and no longer seeing the hangs and long delays as before.

Mike

**Comment by reddevil7nine** [comment]
> 
> 
> @reddevil7nine can you open a specific issue for that? :)

Sorry, just saw this. Thanks @gldraphael
